libraries/standard/mqtt/include/mqtt.h

```diff
@@ -247,7 +247,12 @@ MQTTStatus_t MQTT_Unsubscribe( MQTTContext_t * pContext,
 MQTTStatus_t MQTT_Disconnect( MQTTContext_t * pContext );
 
 /**
- * @brief Loop to receive packets from the transport interface.
+ * @brief Loop to receive packets from the transport interface. Handles keep
+ * alive.
+ *
+ * @note This function assumes that pContext->callbacks.getTime is defined.
```

Comment:
Note that we are assuming `pContext->callbacks.getTime != NULL` here, but there are lots of other places in this code where we're checking this at runtime.  Why is it handled differently in this function?

Reply:
The different between `MQTT_ProcessLoop` and `MQTT_ReceiveLoop` is that the process loop handles keep alive. We need a time implementation for this, but don't otherwise.